src/client/Microsoft.Identity.Client/Instance/Discovery/InstanceDiscoveryManager.cs

```diff
@@ -123,17 +122,32 @@ internal class InstanceDiscoveryManager : IInstanceDiscoveryManager
                 case AuthorityType.Aad:
 
                     var entry = _userMetadataProvider?.GetMetadataOrThrow(environment, requestContext.Logger);
-                    
+
                     if (entry == null && forceValidation)
                     {
                         // only the network provider does validation
                         await FetchNetworkMetadataOrFallbackAsync(requestContext, authorityUri).ConfigureAwait(false);
                     }
-                    
+
+                    // previous
                     entry = entry ??
                         await _regionDiscoveryProvider.GetMetadataAsync(authorityUri, requestContext).ConfigureAwait(false) ??
                         await FetchNetworkMetadataOrFallbackAsync(requestContext, authorityUri).ConfigureAwait(false);
-                                     
+
+                    // new
+                    //if (entry == null)
+                    //{
+                    //    entry = await FetchNetworkMetadataOrFallbackAsync(requestContext, authorityUri).ConfigureAwait(false);
+                    //    if (!string.IsNullOrEmpty(requestContext.ServiceBundle.Config.AzureRegion))
```

Comment:
>Config.AzureRegion [](http://example.com/codeflow?start=81&length=18)

we always did .GetMetadataAsync before? i'm glad you're checking for the region first now

Reply:
Sorry it's confusing. It's actually the opposite. Uncommented one is the current code, we check regional first now. Either way `FetchNetworkMetadataOrFallbackAsync` actually still ends up checking metadata cache first.